### DISCUSSION TITLE
Tacotron2 bad test synthesis results

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#138] [Archived]
<USER>>>> fatih_kiralioglu<br/> <DATE> [February 4, 2020, 2:13pm]<br/> <br/> Hello, *slash* <br/> I have tried a Tacotron2 training with a custom configuration using the<br/> my own voice database *slash* <br/> Unfortunately, the test synthesis results are not good and only 2 out of<br/> 4 test synthesis records are intelligible. *slash* <br/> I guess that there is a very basic error causing this behavior but I<br/> couldn't figure it out. *slash* <br/> I hare the config file content below.<br/> <br/> I also note that I could get very good synthesis results with Tacotron1<br/> method and default config<br/> <br/> Any help is appreciated. *slash* <br/> Best Regards.<br/> <br/> { *slash* <br/> 'github_branch':'inside_docker', *slash* <br/> 'restore_path': *slash* '/mnt/lvm/MozillaTTS/english_tts/TTS-master-gpu_melissa_2/keep/ljspeech-January-10-2020_09+47AM-0000000/checkpoint_50000.pth.tar *slash* ', *slash* <br/> 'run_name': 'ljspeech', *slash* <br/> 'run_description': 'Tacotron2 ljspeech release training',<br/> <br/> 'audio':{<br/> // Audio processing parameters<br/> 'num_mels': 80, // size of the mel spec frame.<br/> 'num_freq': 1025, // number of stft frequency levels. Size of the linear spectogram frame.<br/> 'sample_rate': 16000, // DATASET-RELATED: wav sample-rate. If different than the original data, it is resampled.<br/> 'frame_length_ms': 50, // stft window length in ms.<br/> 'frame_shift_ms': 12.5, // stft window hop-lengh in ms.<br/> 'preemphasis': 0.98, // pre-emphasis to reduce spec noise and make it more structured. If 0.0, no -pre-emphasis.<br/> 'min_level_db': -100, // normalization range<br/> 'ref_level_db': 20, // reference level db, theoretically 20db is the sound of air.<br/> 'power': 1.5, // value to sharpen wav signals after GL algorithm.<br/> 'griffin_lim_iters': 60,// #griffin-lim iterations. 30-60 is a good range. Larger the value, slower the generation.<br/> // Normalization parameters<br/> 'signal_norm': true, // normalize the spec values in range [0, 1]<br/> 'symmetric_norm': true, // move normalization to range [-1, 1]<br/> 'max_norm': 4, // scale normalization to range [-max_norm, max_norm] or [0, max_norm]<br/> 'clip_norm': true, // clip normalized values into the range.<br/> 'mel_fmin': 60.0, // minimum freq level for mel-spec. ~50 for male and ~95 for female voices. Tune for dataset!!<br/> 'mel_fmax': 8000.0, // maximum freq level for mel-spec. Tune for dataset!!<br/> 'do_trim_silence': false // enable trimming of slience of audio as you load it. LJspeech (false), TWEB (false), Nancy (true)<br/> },<br/> <br/> 'distributed':{<br/> 'backend': 'nccl',<br/> 'url': 'tcp: *slash* / *slash* /localhost:54321'<br/> },<br/> <br/> 'reinit_layers': [],<br/> <br/> 'model': 'Tacotron2', // one of the model in models/ <br/> 'grad_clip': 1, // upper limit for gradients for clipping.<br/> 'epochs': 1000, // total number of epochs to train.<br/> 'lr': 0.0001, // Initial learning rate. If Noam decay is active, maximum learning rate.<br/> 'lr_decay': false, // if true, Noam learning rate decaying is applied through training.<br/> 'warmup_steps': 4000, // Noam decay steps to increase the learning rate from 0 to 'lr'<br/> 'memory_size': 5, // ONLY TACOTRON - size of the memory queue used fro storing last decoder predictions for auto-regression. If < 0, memory queue is disabled and decoder only uses the last prediction frame.<br/> 'attention_norm': 'softmax', // softmax or sigmoid. Suggested to use softmax for Tacotron2 and sigmoid for Tacotron.<br/> 'prenet_type': 'original', // 'original' or 'bn'.<br/> 'prenet_dropout': true, // enable/disable dropout at prenet.<br/> 'windowing': false, // Enables attention windowing. Used only in eval mode.<br/> 'use_forward_attn': true, // if it uses forward attention. In general, it aligns faster.<br/> 'forward_attn_mask': false,<br/> 'transition_agent': false, // enable/disable transition agent of forward attention.<br/> 'location_attn': false, // enable_disable location sensitive attention. It is enabled for TACOTRON by default.<br/> 'loss_masking': true, // enable / disable loss masking against the sequence padding.<br/> 'enable_eos_bos_chars': false, // enable/disable beginning of sentence and end of sentence chars.<br/> 'stopnet': true, // Train stopnet predicting the end of synthesis.<br/> 'separate_stopnet': true, // Train stopnet seperately if 'stopnet==true'. It prevents stopnet loss to influence the rest of the model. It causes a better model, but it trains SLOWER.<br/> 'tb_model_param_stats': false, // true, plots param stats per layer on tensorboard. Might be memory consuming, but good for debugging.<br/> <br/> 'batch_size': 32, // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.<br/> 'eval_batch_size':16, <br/> 'r': 1, // Number of decoder frames to predict per iteration. Set the initial values if gradual training is enabled. <br/> 'gradual_training': null, //[[0, 7, 32], [1, 5, 32], [50000, 3, 32], [230000, 2, 16], [390000, 1, 8]], // ONLY TACOTRON - set gradual training steps [first_step, r, batch_size]. If it is null, gradual training is disabled. <br/> 'wd': 0.000001, // Weight decay weight.<br/> 'checkpoint': true, // If true, it saves checkpoints per 'save_step'<br/> 'save_step': 10000, // Number of training steps expected to save traninpg stats and checkpoints.<br/> 'print_step': 25, // Number of steps to log traning on console.<br/> 'batch_group_size': 0, //Number of batches to shuffle after bucketing.<br/> <br/> 'run_eval': true,<br/> 'test_delay_epochs': 5, //Until attention is aligned, testing only wastes computation time.<br/> 'test_sentences_file': null, // set a file to load sentences to be used for testing. If it is null then we use default english sentences.<br/> 'min_seq_len': 5, // DATASET-RELATED: minimum text length to use in training<br/> 'max_seq_len': 150, // DATASET-RELATED: maximum text length<br/> 'output_path': 'keep/', // DATASET-RELATED: output path for all training outputs.<br/> 'num_loader_workers': 8, // number of training data loader processes. Don't set it too big. 4-8 are good values.<br/> 'num_val_loader_workers': 4, // number of evaluation data loader processes.<br/> 'phoneme_cache_path': 'mozilla_us_phonemes', // phoneme computation is slow, therefore, it caches results in the given folder.<br/> 'use_phonemes': true, // use phonemes instead of raw characters. It is suggested for better pronounciation.<br/> 'phoneme_language': 'en-us', // depending on your target language, pick one from https://github.com/bootphon/phonemizer#languages<br/> 'text_cleaner': 'phoneme_cleaners',<br/> 'use_speaker_embedding': false, // use speaker embedding to enable multi-speaker learning.<br/> 'style_wav_for_test': null, // path to style wav file to be used in TacotronGST inference.<br/> 'use_gst': false, // TACOTRON ONLY: use global style tokens<br/> <br/> 'datasets': // List of datasets. They all merged and they get different speaker_ids.<br/> [<br/> {<br/> 'name': 'ljspeech',<br/> 'path': '/mnt/lvm/MozillaTTS/english_tts/melissa_all_trimmed',<br/> 'meta_file_train': 'metadata_train.csv',<br/> 'meta_file_val': 'metadata_val.csv'<br/> }<br/> ]<br/> <br/> }<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/tacotron2-bad-test-synthesis-results]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [March 1, 2020, 11:20am]<br/> <br/> Try finetuning the model with forward attention and batch normalization,<br/> instead. It has been trained for a large number of iterations and will<br/> be able to align to the new voice much faster, even if the data amount<br/> is smaller.<br/> <br/> [Archived Post]<br/>
